FV3/atmos_cubed_sphere/tools/coarse_grained_restart_files.F90

```diff
@@ -0,0 +1,366 @@
+       coarse_domain, fine_bd, coarse_bd, restart)
+    integer, intent(in) :: tile_count, nz, nt_prog, nt_phys
+    logical, intent(in) :: hydrostatic, hybrid_z, agrid_vel_rst, fv_land
+    type(domain2d), intent(inout) :: coarse_domain
+    type(fv_grid_bounds_type), intent(in) :: fine_bd
+    type(fv_coarse_grid_bounds_type), intent(in) :: coarse_bd
+    type(coarse_restart_type), intent(inout) :: restart
+
+    call get_fine_array_bounds(fine_bd, is, ie, js, je)
+    call get_coarse_array_bounds(coarse_bd, is_coarse, ie_coarse, js_coarse, je_coarse)
+    n_prognostic_tracers = nt_prog
+    n_diagnostic_tracers = nt_phys
+    n_tracers = nt_prog + nt_phys
+    npz = nz
+
+    call allocate_coarse_restart_type(hydrostatic, hybrid_z, &
+         agrid_vel_rst, fv_land, restart)
+    call register_coarse_restart_files(tile_count, hydrostatic, &
+         hybrid_z, agrid_vel_rst, fv_land, &
+         coarse_domain, restart)
+  end subroutine fv_coarse_restart_init
+
+  subroutine fv_io_write_restart_coarse(Atm, grids_on_this_pe, timestamp)
+    type(fv_atmos_type), intent(inout) :: Atm(:)
+    logical, intent(in) :: grids_on_this_pe(:)
+    character(len=*), optional, intent(in) :: timestamp
+
+    integer :: tile_count, n_tiles
+
+    n_tiles = size(Atm(:))
+
+    do tile_count = 1, n_tiles
+       if (.not. grids_on_this_pe(tile_count)) cycle
+       call coarse_grain_restart_data(Atm(tile_count))
+       call save_restart(Atm(tile_count)%coarse_graining%restart%fv_core_coarse, timestamp)
+       call save_restart(Atm(tile_count)%coarse_graining%restart%fv_tracer_coarse, timestamp)
+       call save_restart(Atm(tile_count)%coarse_graining%restart%fv_srf_wnd_coarse, timestamp)
+       if (Atm(tile_count)%flagstruct%fv_land) then
+          call save_restart(Atm(tile_count)%coarse_graining%restart%mg_drag_coarse, timestamp)
+          call save_restart(Atm(tile_count)%coarse_graining%restart%fv_land_coarse, timestamp)
+       endif
+    enddo
+  end subroutine fv_io_write_restart_coarse
+
+  subroutine allocate_coarse_restart_type(hydrostatic, hybrid_z, agrid_vel_rst, fv_land, restart)
+    logical, intent(in) :: hydrostatic, hybrid_z, agrid_vel_rst, fv_land
+    type(coarse_restart_type), intent(inout) :: restart
+
+    allocate(restart%u(is_coarse:ie_coarse,js_coarse:je_coarse+1,npz))
+    allocate(restart%v(is_coarse:ie_coarse+1,js_coarse:je_coarse,npz))
+    allocate(restart%u_srf(is_coarse:ie_coarse,js_coarse:je_coarse))
+    allocate(restart%v_srf(is_coarse:ie_coarse,js_coarse:je_coarse))
+    allocate(restart%delp(is_coarse:ie_coarse,js_coarse:je_coarse,npz))
+    allocate(restart%pt(is_coarse:ie_coarse,js_coarse:je_coarse,npz))
+    allocate(restart%q(is_coarse:ie_coarse,js_coarse:je_coarse,npz,n_prognostic_tracers))
+    allocate(restart%qdiag(is_coarse:ie_coarse,js_coarse:je_coarse,npz,n_prognostic_tracers+1:n_tracers))
+    allocate(restart%phis(is_coarse:ie_coarse,js_coarse:je_coarse))
+
+    if (agrid_vel_rst) then
```

Comment:
Nice to have this option!

Reply:
It was already built into the model!  

It's just for "data-assimilation" purposes however (I'm not sure exactly what this means), so there's currently no functionality for actually using these winds to initialize the model.

https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/42f2877a4e04882a1046d35662bce61c85385e27/FV3/atmos_cubed_sphere/model/fv_arrays.F90#L903-L906

Reply:
Got it, well hopefully we will be able to repurpose the existing A->D grid transformation routine that Lucas pointed out!